### PR TITLE
Tracker: get game names from slot_info instead of `multidata["games"]` and render custom game names on generic tracker

### PR DESCRIPTION
**Pull request by alwaysintreble**
## What is this fixing or adding?
https://discord.com/channels/731205301247803413/731214280439103580/1075228226399502366

## How was this tested?
ran webhost locally and opened the multitracker

### CODE REVIEW

**Comment by alwaysintreble** [comment]
Fixed the multitracker, but the individual trackers are currently determined by figuring out which game each player is using from `multidata["games"]` so we need a way to determine the players' games. Only solution i have offhand is making it part of an api endpoint like the one I was too lazy to finish.

**Comment by alwaysintreble** [comment]
Pulling the game names from `slot_info` and falling back to `games` just in case. While I was in there got custom datapackage games to render on the player tracker by updating the lookup dictionary. Tested by generating on my Messenger branch after merging current main into it, then running WebHost on this branch where the game doesn't exist.
Screenshot_39

File: WebHostLib/tracker.py [hunk 1]
```diff
@@ -260,8 +260,18 @@ def get_static_room_data(room: Room):
     names: Dict[int, Dict[int, str]] = multidata["names"]
     groups = {}
     if "slot_info" in multidata:
+        games = {slot: slot_info.game for slot, slot_info in multidata["slot_info"].items()}
         groups = {slot: slot_info.group_members for slot, slot_info in multidata["slot_info"].items()
                   if slot_info.type == SlotType.group}
+
+        for game in games.values():
+            if game in multidata["datapackage"]:
+                lookup_any_location_id_to_name.update(
```
Comment (Berserker66) [inline WebHostLib/tracker.py:269]: these are globals and shared between multiple Rooms, which can be problematic.

Comment (alwaysintreble) [reply WebHostLib/tracker.py:269]: I was originally using a completely separate dictionary and updating that with the new names, then using `any.get(id, custom.get(id, id))`. should i just create a copy of the global and update that instead? seems better.

Comment (Berserker66) [reply WebHostLib/tracker.py:269]: I'd recommend keeping the global lookup and then also extracting the special data as static room data, then using chain() for lookups.

Comment (alwaysintreble) [reply WebHostLib/tracker.py]: looks like the custom trackers don't use the filter at all, so i build the filter right before rendering of the generic tracker. couldn't figure out how to pass it as an arg from searching. did check that the multi and all the custom trackers still work with this change though 🤷

File: WebHostLib/tracker.py [hunk 2]
```diff
@@ -1207,6 +1219,9 @@ def __renderGenericTracker(multisave: Dict[str, Any], room: Room, locations: Dic
     for order_index, networkItem in enumerate(ordered_items, start=1):
         player_received_items[networkItem.item] = order_index
 
+    app.jinja_env.filters["location_name"] = lambda loc: collections.ChainMap(lookup_any_location_id_to_name, custom_locations).get(loc, loc)
+    app.jinja_env.filters["item_name"] = lambda item: collections.ChainMap(lookup_any_item_id_to_name, custom_items).get(item, item)
```
Comment (Berserker66) [inline WebHostLib/tracker.py:1223]: this still sets the global filter right before rendering. Jinja can be called in a multithreaded fashion.

**Review by Berserker66 — APPROVED** [review]

**Pull request merged**